### ISSUE TITLE
In V3.2.6 when new is a macro the detection in vector.h can be fooled

### ISSUE THREAD

**Issue by transmitterdan** [wxWidgets/wxWidgets#24994]
### Description

wxVector::push_back() can cause compile time error under MS VS2022 if new is a macro such as in a DEBUG build. The proposed solution by WX is to not include msvcrt.h or to define __NO_VC_CRTDBG__. However, neither of these solutions work if the program defines its own 'new' macro per the MS recommendation.

#### Bug description:

At [line 67](https://github.com/wxWidgets/wxWidgets/blob/5ff25322553c1870cf20a2e1ba6f20ed50d9fe9a/include/wx/vector.h#L67C1-L67C19) the test for whether new is a macro can be fooled. Maybe it's better to just test if new is defined or maybe just unilaterally undef new.

So in vector.h we could just say this:
```#ifdef new```
insteadl of:
```#ifdef WXDEBUG_NEW```

### Platform and version information

- wxWidgets version  3.2.6
- wxWidgets port wxMSW
- OS Windows 11

**Comment by transmitterdan** [comment]
Yes, it's a terrible idea, but Microsoft does recommend it. And their debugger has embedded code that can dump orphan blocks from the heap if they were allocated by using the macro version of new.  We could check if new is a macro and undef it at the top of vector.h. Then at the end we could check if WXDEBUG_NEW is defined and if it is, then redefine new as WXDEBUG_NEW. This would probably not break any existing wxWidgets projects that use the msvcrt.h include file.  In fact, it may be another bug that vector.h does not now redefine new at the end after it undef-ed it at the beginning.

**Comment by vadz** [comment]
BTW, as long as it's MSVC-specific, we could use `#pragma push_macro("new")` and `#pragma pop_macro("new")` at the end.

**Comment by transmitterdan** [comment]
Yes, that would be the perfect solution for MSVC. I think that is the only compiler where new is typically turned into a macro.

**Comment by vadz** [comment]
I've added a commit doing this to 3.2 proposed fixes branch, please test it if you can, I hope to merge this before 3.2.7.

**Comment by transmitterdan** [comment]
Compiles and runs ok for my test case.
